NestJS/src/gmail_api/gmailapi.js

```diff
@@ -117,43 +121,62 @@ function listLabels(auth) {
 
 }
 
-function sendMessage(auth, data) {
-  const gmail = google.gmail({version: 'v1', auth});
-  var raw = makeBody(data);
- 
-  gmail.users.messages.send({
+function sendMessage(auth, data, recipient) {
+    const gmail = google.gmail({version: 'v1', auth});
+    var raw = makeBody(data, recipient);
+    try{
+  gmail.users.messages.send({ 
       auth: auth,
       userId: 'me',
       resource: {
           raw: raw
       }
   })
+  }
+  catch(error){
+    console.log("Error during authorization " + error)
+  }
 }
 
-function makeBody(data) {
+function makeBody(data, recipient) {
   var boundary = "__myapp__";
   var nl = "\n";
   let fileToAttach = '/home/reedvl/Downloads/test.docx';
   var attach = new Buffer(fs.readFileSync(fileToAttach)) .toString("base64");
 
-  var arrays = defineTypeOfRecipients(data);
-  let To = arrays[0].toString()
-  let CC = arrays[1].toString()
-  let BCC = arrays[2].toString()
+  // var arrays = defineTypeOfRecipients(data);
+  // let To = arrays[0].toString()
+  // let CC = arrays[1].toString()
+  // let BCC = arrays[2].toString()
+  let type
+
+  switch(recipient.type){
+    case "to":
+      type = "to: " + recipient.email
+      break;
+    case "cc":
+      type = "cc: " + recipient.email
+      break;
+    case "bcc":
+      type = "bcc: " + recipient.email
+      break;
+    default:
+      break;
+  }
```

Comment:
why do you use switch? you can just assign.

Reply:
My fault, Thank you!